inference/cluster_store.py

```diff
@@ -82,8 +83,15 @@ def build_optimal_agglomerative_clusters(self):
     # Function to run agglomerative clustering 
     # with specified number of cluster: c -> int
     def build_agglomerative_clusters(self, c):
-        # TODO
-        pass
+        self.total_cluster = c
+        X = np.asarray(self.original_data)
```

Comment:
Has this been tested? I feel like this would break since `self.original_data` is a 2D array.

Reply:
I have tested it while passing a 2D array for data and it worked as expected.